cmd/fetcher/app/server.go

```diff
@@ -85,11 +64,31 @@ func Run(logger *zap.Logger) {
 		}
 	}
 
-	go func() {
-		if err := registerTraceExporter(*collectorEndpoint); err != nil {
-			logger.Fatal("could not register trace exporter", zap.Error(err), zap.String("collector_endpoint", *collectorEndpoint))
+	openTracingEnabled, err := strconv.ParseBool(os.Getenv("OPENTRACING_ENABLED"))
+	if err != nil {
+		logger.Fatal("error parsing OPENTRACING_ENABLED", zap.Error(err))
+	}
+
+	var shutdown func()
+	if openTracingEnabled {
+		go func() {
+			if err := tracing.RegisterTraceExporter(logger, *collectorEndpoint, "Fission-Fetcher"); err != nil {
+				logger.Fatal("could not register trace exporter", zap.Error(err), zap.String("collector_endpoint", *collectorEndpoint))
+			}
+		}()
+	} else {
+		shutdown, err = otelUtils.InitProvider(logger, "Fission-Fetcher")
+		if err != nil {
+			logger.Fatal("error initializing provider for OTLP", zap.Error(err))
 		}
-	}()
+	}
+	if shutdown != nil {
```

Comment:
Should we move this part under else only ? we may not need the shutdown definition above then. Same for rest of shutdown definitions.